storage/include/s3/client.hpp

```diff
@@ -165,9 +165,9 @@ class Client : public concord::storage::IDBClient {
   }
 
   concordUtils::Status multiPut(const SetOfKeyValuePairs& _keyValueMap) override {
-    ITransaction::Guard g(beginTransaction());
-    for (auto&& pair : _keyValueMap) g.txn()->put(pair.first, pair.second);
-
+    std::unique_ptr<ITransaction> txn(beginTransaction());
```

Comment:
Why not use ITransaction::Guard here, given that you have fixed it in case of commit exception?

Reply:
Because I want the exception to be thrown and it's impossible from a ITransaction::~Guard

Reply:
Since I'm performing a commit only if not during uncaught exception, I'd probably should to the following:
virtual ~Guard() **noexcept(false)** {
      if (!std::uncaught_exception()) txn_->commit();
      delete txn_;
    }